src-ui/lib/openLibrary.spec.ts

```diff
@@ -26,6 +26,25 @@ describe('book', () => {
       }
       expect(book).toEqual(expected)
     })
+    it('should throw a timeout error when Open Library is down and request times out', async () => {
+      global.fetch = vi.fn(
+        (_: Request, { signal }: RequestInit) =>
+          new Promise<Response>((_resolve, reject) => {
+            // Simulate the request being aborted
+            signal?.addEventListener('abort', () => {
+              reject(new DOMException('Aborted', 'AbortError'))
+            })
+          })
+      )
+      await getByISBN(isbn)
+        .then(() => {
+          throw new Error('Test failed: Expected a timeout error')
+        })
+        .catch((error) => {
+          expect(error.message).toBe('Fetch request timed out')
+        })
```

Comment:
This test only checks that you handle the `abort` error correctly -- not that the timeout itself works.